_posts/2024-12-02-ranger.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: post
+title: "The long journey to Apache Ranger"
+author: "Manfred Moser"
+excerpt_separator: <!--more-->
+image: /assets/images/logos/apache-ranger.png
+---
+
+[Apache Ranger]({{site.baseurl}}/ecosystem/add-on.html#apache-ranger) has
+arrived! With the new [Trino
+466]({{site.baseurl}}/docs/current/release/release-466.html) you all get another
+jam-packed release of Trino awesomeness. One of the goodies is a new plugin for
+access control for your data with Apache Ranger, and it has gone through a long
+story to get here.
+
+<!--more-->
+
+## The beginnings
+
+Apache Ranger has been in use with Trino for a long time - in fact there are
+[early](https://github.com/trinodb/trino/pull/244)
+[rudimentary](https://github.com/trinodb/trino/pull/1069) pull requests from
+2019 that implemented some support. And even before then various hacks existed.
+In 2020 a plugin for PrestoSQL was added to Apache Ranger. Aakash Nand blogged
+about [Integrating Trino and Apache
+Ranger](https://towardsdatascience.com/integrating-trino-and-apache-ranger-b808f6b96ad8)
+in 2021 to adjust for the changes to Trino. Jeff Xu followed up with
+[Integrating Trino and Apache Ranger in a Kerberos secured enterprise
+environment](https://medium.com/@jeff.xu.z/integrating-trino-and-apache-ranger-in-a-kerberos-secured-enterprise-environment-997c95cd10e9)
+in 2022, followed quickly by the addition of the Trino support to the Apache
+Ranger repository.
+
+## Testing and container images
+
+However that was only half of the needed support. The Trino project moves very
+fast with nearly weekly releases, so the best approach is to have the supporting
+plugin in Trino directly so every release includes the relevant updates. [Erik
+Anderson](https://github.com/dprophet) created a more mature plugin that was in
+production use for quite a while for Trino. His [pull request from July
+2022](https://github.com/trinodb/trino/pull/13297) included great background
+reasoning for having the plugin in Trino. One of the issues that Erik solved for
+the Trino project is testing. Trino plugins require the availability of a
+container image for testing whatever integration. Apache Ranger did still not
+ship a container in 2022, but thanks to the lobbying efforts of Erik this
+changed and a container image became available over the months.
+
+## A long sprint
+
+Unfortunately focus changed and while the PR from Erik existed and was useful,
+it never made it to merge due to waning priorities. That changed when [Madhan
+Neethiraj](https://github.com/mneethiraj) from the Apache Ranger project stepped
+up and created [new PR](https://github.com/trinodb/trino/pull/22675) in July 2024. 
+
+We knew this could be another shot at it and it would require a lot of work to
+get it done, since we put a high focus on quality so that we can maintain the
+Trino codebase for the long run. Monitoring all PRs regularly [Manfred
```

Comment:
```suggestion
Trino codebase for the long run. Monitoring all PRs regularly, [Manfred
```
If this blog is written by you, you can say "I" here instead of Manfred Moser
Or maybe I (Manfred Moser) if you want to keep your link